app/src/lib/git/stage.ts

```diff
@@ -24,47 +24,50 @@ export async function stageManualConflictResolution(
   repository: Repository,
   file: WorkingDirectoryFileChange,
   manualResolution: ManualConflictResolution
-): Promise<boolean> {
+): Promise<void> {
   const { status } = file
   // if somehow the file isn't in a conflicted state
   if (!isConflictedFileStatus(status)) {
     log.error(`tried to manually resolve unconflicted file (${file.path})`)
-    return false
+    return
   }
   if (!isManualConflict(status)) {
     log.error(
       `tried to manually resolve conflicted file with markers (${file.path})`
     )
-    return false
+    return
   }
 
   const chosen =
     manualResolution === ManualConflictResolutionKind.theirs
       ? status.entry.them
       : status.entry.us
 
-  let exitCode: number = -1
-
   switch (chosen) {
     case GitStatusEntry.Deleted: {
-      exitCode = (await git(
-        ['rm', file.path],
-        repository.path,
-        'removeConflictedFile'
-      )).exitCode
+      await git(['rm', file.path], repository.path, 'removeConflictedFile')
+      break
+    }
+    case GitStatusEntry.Added: {
+      await git(['add', file.path], repository.path, 'addConflictedFile')
       break
     }
-    case GitStatusEntry.Added:
     case GitStatusEntry.UpdatedButUnmerged: {
-      exitCode = (await git(
-        ['add', file.path],
+      const choiceFlag =
+        manualResolution === ManualConflictResolutionKind.theirs
+          ? 'theirs'
+          : 'ours'
+      const checkoutCompleted = await git(
+        ['checkout', `--${choiceFlag}`, '--', file.path],
         repository.path,
-        'addConflictedFile'
-      )).exitCode
+        'checkoutConflictedFile'
+      )
+      if (checkoutCompleted) {
```

Comment:
I belive we can drop this check now as well given that the previous `checkout` invocation will throw an exception if it exits with a non-zero exit code.

Reply:
ah good catch!